### PR TITLE
Plat 8781

### PR DESCRIPTION
**Pull request by leandrobono**
## Ticket

PLAT-8781
_Related tickets:_
_Related PRs:_

## Type of PR

- [X] Bugfix
- [ ] New feature
- [ ] Minor changes

## Running the demo

In order to run the included demo app, run `yarn run bootstrap` and then `cd demo` and run both `npx react-native run-android` and `npx react-native start` in parallel.

## Review checklist for new or updated modules

Please go through the following checks:

- [ ] I have made no modules changes - ignore this section questions in that case
- [ ] I have included a `meta.json` file in the module's root directory
- [ ] I have included a `preview.png` image in the module's root directory
- [ ] `yarn run parse` checks pass
- [X] I updated the modules build (`/dist` directory) with `yarn run dist`

React Native modules specific checks:

- [ ] `yarn run demo` and `yarn run add app-menu <your-module>` runs correctly
- [ ] **"Running the demo"** app shows the App Menu screen with an entry for your module
- [ ] Clicking your module entry in the menu loads your module's initial screen correctly

Django modules specific checks:

- [ ] `python manage.py show_urls` shows that the module's urls are resolved correctly
- [ ] I have checked that the migrations are detected and run correctly
- [ ] I have checked that the admin is working correctly for the module

If your module is a "supermodule" and therefore contains both React Native and Django code, then please go through all of the checks above.

## Review checklist for scaffold changes

- [ ] I have made no scaffold changes - ignore this section questions in that case
- [ ] `yarn run template` produces no changes
- [ ] `yarn run raw && yarn run cookie` produces no changes, except for `yarn.lock`
- [ ] `yarn run demo` finishes successfully, `demo` directory gets created locally
- [ ] **"Running the demo"** opens the app correctly in the welcome screen

## Changes introduced
This fix separate the upload_to_play_store function call on the internal track from the check of first internal deployment.

## Test and review

To test this PR, run the deployment when you dont have everything set on Google Play Console, and you must see the error on Circle and we wont upload anything to google, just build. When everything is set, deployment must work as always.

### CODE REVIEW

**Review by math-a3k — DISMISSED** [review]
They look good to me - although not my area, I would be nice if someone who is more into the mobile deployments take a look

**Review by alaisgomes — APPROVED** [review]

**Review by GaboGomezT — APPROVED** [review]

File: scaffold/template/custom/android/fastlane/Fastfile [hunk 1]
```diff
@@ -32,20 +32,22 @@ platform(:android) do
 
   desc('Submit a new Internal Build')
   lane(:internal) do
+    upload = true
     begin
       internal_files = google_play_track_version_codes(track:"internal")
-
-      internal_files.length > 0
-
+    rescue => ex
```
Comment (mabdullahabid) [inline scaffold/template/custom/android/fastlane/Fastfile:38]: When does `rescue => ex` code get executed? Basically, how and when will `upload = false` be executed?

**Review by mabdullahabid — APPROVED** [review]
Code-wise it looks good to me. Just a question on the execution logic. Also, would be nice if we can get input from @danielsousaio on the Ruby code.

**Comment by mabdullahabid** [comment]
@leandrobono also, please add QA steps

File: scaffold/template/custom/android/fastlane/Fastfile (hunk 1 above)
Comment (leandrobono) [reply scaffold/template/custom/android/fastlane/Fastfile:38]: this gets executed when theres nothing uploaded and code throws exception

**Review by danielsousaio — APPROVED** [review]

**Pull request merged**